.github/scripts/changelog-generator-v2.js

```diff
@@ -0,0 +1,88 @@
+const { Octokit } = require('octokit')
+const fs = require('fs')
+const { Parser } = require('json2csv')
+
+const octokit = new Octokit({
+	auth: process.env.GITHUB_TOKEN
+})
+
+const { baseBranch, headTag, repositoryName: repo, owner } = process.env
+
+const filters = process.env.filters.split(',')
+
+// Fetch release details
+async function fetchReleaseDetails () {
+	const release = await octokit.rest.repos.getReleaseByTag({
+		owner,
+		repo,
+		tag: headTag
+	})
+	return new Date(release.data.created_at)
+}
+
+// Fetch the last 100 commits
+async function fetchAllCommits (releaseDate) {
+	const { data: commits } = await octokit.rest.repos.listCommits({
+		owner,
+		repo,
+		per_page: 100
+	})
+
+	return commits.filter(commit => {
+		const commitDate = new Date(commit.commit.committer.date)
```

Comment:
Why do you need the date? you need the diff between the baseBranch and the headTag

Reply:
Squash merging will affect the results of compareCommits.

In a squash merge scenario, these two methods are likely to yield different results because the "compareCommits" will consider the single squash commit while the "Filtering commits by date" won't see the individual commits that were included in the squash if they were made before the date you're filtering from.